src/openforms/forms/models/form.py

```diff
@@ -380,12 +397,27 @@ def __str__(self):
     @property
     def is_available(self) -> bool:
         """
-        Soft deleted, deactivated or forms in maintenance mode are not available.
+        Soft deleted, deactivated, forms in maintenance mode or forms which have reached the
+        submission limit are not available.
         """
-        if any((self._is_deleted, not self.active, self.maintenance_mode)):
+        if any(
+            (
+                self._is_deleted,
+                not self.active,
+                self.maintenance_mode,
+                self.has_reached_submissions_limit(),
+            )
+        ):
             return False
         return True
 
+    def has_reached_submissions_limit(self) -> bool:
+        if (
+            limit := self.submission_maximum_allowed
+        ) and limit == self.submission_counter:
```

Comment:
This check is duplicated in many places - you should use this property instead of repeating the equality check, and as mentioned before, check for greater than or equal rather than just equality.